examples/chart/teleport-cluster/values.yaml

```diff
@@ -30,6 +30,10 @@ authenticationSecondFactor:
 # Possible values are 'multiplex'
 proxyListenerMode: ""
 
+# Re-configures the teleport kubernetes proxy to listen on a different public address/port rather than the default 
+# cluster address (e.g. k8s.example.com:443 instead of teleport.example.com:3026)
+proxyKubePublicAddr: ""
```

Comment:
I'm not a huge fan of this approach because it doesn't automatically configure a load balancer or anything else to make this actually work - it just expects the user to do all that work manually.

If we were to expose this value, it would at least need to automatically configure an extra load balancer service with a separate public IP. Currently I think users will just find it confusing that they've specified a different hostname but it doesn't work out of the box.

Reply:
Hm thats makes sense! And it would actually be lovely to have that set up in the chart since it's basically what we are trying to do (but currently we're doing it through terraform, ie setting the chart load balancer to cluster IP and creating two load balancers - 1 for the ssh + proxy etc and 1 for the kubernetes endpoint).. we are in addition trying to configure a set of IPs to be whitelisted via loadBalancerSourceRanges on the service.

Would it make sense if we implemented this on the chart?
- ability to configure a **kubePublicAddr** separately
  - this would in turn:
  - configure two load balancers instead of just one 
    - 1 for all ports but kubernetes
    - 1 for kubernetes, with optionally setting a list of ranges to be whitelisted via loadBalancerSourceRanges (eg kubeWhitelistedRanges)

Please let us know your thoughts and we can get something implemented for review!

Cheers :)

Reply:
Hey - I have to say that I don't think we would look to merge such a contribution into the upstream chart. This is because it adds a reasonable amount of additional complexity to the chart that's only currently desired by one user - i.e. yourself. 

If we were to implement this for Kubernetes, why would we not also do it for the database address, the tunnel address, the SSH public address etc? Each of these would be an extra value to set and an extra conditional to evaluate - I don't think it's very scalable to do this when we haven't had previous requests for this functionality.

I totally understand your desire to have this functionality for your own use case, but I would encourage you to implement this functionality on your own fork of the chart instead!

Reply:
Hi! I completely agree with your points and understand it would add a fair amount of complexity to the chart that might not balance with the incoming usage requests at the moment. As for the first item, do you think it's still something that would make sense to get its way into chart? (ie kubePublicAddr setting up a separate load balancer, which would probably require some load balancer config associated with it)

thanks!

Reply:
Hey - it's the same thing I'm afraid, as it's only useful to you (at the moment) I wouldn't accept this for the upstream chart :(